src/docs/stan-reference/examples.tex

```diff
@@ -7000,12 +7000,10 @@ \section{Simulating from a Gaussian Process}
   real x[N];
 }
 transformed data {
-  vector[N] mu;
-  cov_matrix[N] K;
-  mu = rep_vector(0, N);
-  K = cov_exp_quad(x, 1, 1);
-  for (i in 1:N)
-    K[i, i] = K[i, i] + 0.1;
+  matrix[N, N] K = cov_exp_quad(x, 1.0, 1.0);
```

Comment:
I guess it'd be fine to use `cov_matrix` in the data block because the constraint checking will only get called once. I never use `cov_matrix` in a model because I'm usually generating the covariance matrix in the `model` block or in a local block in the `transformed parameters` block. If we were to have a covariance matrix defined in the `transformed parameter` I would strongly advocate against using `cov_matrix` because K + diag_matrix(1e-9) will be positive definite by construction and checking covariance matrix constraint checking is redundant.